packages/dart_client/lib/src/db/chat_persistence_client.dart

```diff
@@ -0,0 +1,212 @@
+
+///
+abstract class ChatPersistenceClient {
+  /// Creates a new connection to the client
+  Future<void> connect(String userId);
+
+  /// Closes the client connection
+  /// If [flush] is true, the data will also be deleted
+  Future<void> disconnect({bool flush = false});
+
+  /// Get stored replies by messageId
+  Future<List<Message>> getReplies(
+    String parentId, {
+    PaginationParams options,
+  });
+
+  /// Get stored connection event
+  Future<Event> getConnectionInfo();
+
+  /// Get stored lastSyncAt
+  Future<DateTime> getLastSyncAt();
+
+  /// Update stored connection event
+  Future<void> updateConnectionInfo(Event event);
+
+  /// Update stored lastSyncAt
+  Future<void> updateLastSyncAt(DateTime lastSyncAt);
+
+  /// Get the channel cids saved in the offline storage
+  Future<List<String>> getChannelCids();
+
+  ///
+  Future<ChannelModel> getChannelByCid(String cid);
+
+  ///
+  Future<List<Member>> getMembersByCid(String cid);
+
+  ///
+  Future<List<Read>> getReadsByCid(String cid);
+
+  ///
+  Future<List<Message>> getMessagesByCid(
+    String cid, {
+    PaginationParams messagePagination,
+  });
+
+  /// Get channel data by cid
+  Future<ChannelState> getChannelStateByCid(
+    String cid, {
+    PaginationParams messagePagination,
+  }) async {
+    final members = await getMembersByCid(cid);
+    final reads = await getReadsByCid(cid);
+    final channel = await getChannelByCid(cid);
+    final messages = await getMessagesByCid(
+      cid,
+      messagePagination: messagePagination,
+    );
+    return ChannelState(
+      members: members,
+      read: reads,
+      messages: messages,
+      channel: channel,
+    );
+  }
+
+  /// Get list of channels by filter, sort and paginationParams
+  Future<List<ChannelState>> getChannelStates({
+    Map<String, dynamic> filter,
+    List<SortOption> sort = const [],
+    PaginationParams paginationParams,
+  });
+
+  /// Update list of channel queries
+  /// If [clearQueryCache] is true before the insert
+  /// the list of matching rows will be deleted
+  Future<void> updateChannelQueries(
+    Map<String, dynamic> filter,
+    List<String> cids,
+    bool clearQueryCache,
+  );
+
+  /// Remove a message by message id
+  Future<void> deleteMessageById(String messageId) {
+    return deleteMessageByIds([messageId]);
+  }
+
+  /// Remove a message by message ids
+  Future<void> deleteMessageByIds(List<String> messageIds);
+
+  /// Remove a message by channel cid
+  Future<void> deleteMessageByCid(String cid) {
+    return deleteMessageByCids([cid]);
+  }
+
+  /// Remove a message by message cids
+  Future<void> deleteMessageByCids(List<String> cids);
+
+  /// Remove a channel by cid
+  Future<void> deleteChannels(List<String> cids);
+
+  /// Update messages data from a list
+  Future<void> updateMessages(String cid, List<Message> messages);
+
+  /// Get the info about channel threads
+  Future<Map<String, List<Message>>> getChannelThreads(String cid);
+
+  ///
```

Comment:
Can you add a simple one-liner to these functions? We can work on this together if you like 😄